code/lib/cli/src/check.ts

```diff
@@ -0,0 +1,72 @@
+import { join } from 'node:path';
+import prompts from 'prompts';
+import { dedent } from 'ts-dedent';
+import execa from 'execa';
+import { readJSON } from 'fs-extra';
+
+function tryFindCore() {
+  try {
+    return require.resolve('@storybook/core');
```

Comment:
I think this check is not enough.

I just ran the canary in a sandbox with pnpm, in it's default mode.
I manually upgraded the project to us this canary, and did NOT run the upgrade command.

This code detected `@storybook/core` present... but it had not been hoisted to the top of the project by pnpm.

So though most code got `@storybook/core` and was able to function... the manager builder (esbuild) failed to compile.

```
> todos-app@0.1.0 storybook /Users/me/Projects/Storybook/repros/todos-app
> storybook dev -p 6006

@storybook/cli v0.0.0-pr-28288-sha-868da8f7

info => Starting manager..
✘ [ERROR] Could not resolve "@storybook/core/core-server/presets/common-manager"

    node_modules/.cache/storybook/1c3385a5d25e538d10b518b310c74d3ca2690b6aaffeadccd74da79736171f86/sb-manager/storybook-core-core-server-presets-0/common-manager-bundle.js:1:7:
      1 │ import '@storybook/core/core-server/presets/common-manager';
        ╵        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  You can mark the path "@storybook/core/core-server/presets/common-manager" as external to exclude
  it from the bundle, which will remove this error and leave the unresolved path in the bundle.

Error: Build failed with 1 error:
node_modules/.cache/storybook/1c3385a5d25e538d10b518b310c74d3ca2690b6aaffeadccd74da79736171f86/sb-manager/storybook-core-core-server-presets-0/common-manager-bundle.js:1:7: ERROR: Could not resolve "@storybook/core/core-server/presets/common-manager"
    at failureErrorWithLog (./node_modules/.pnpm/esbuild@0.20.2/node_modules/esbuild/lib/main.js:1651:15)
    at ./node_modules/.pnpm/esbuild@0.20.2/node_modules/esbuild/lib/main.js:1059:25
    at runOnEndCallbacks (./node_modules/.pnpm/esbuild@0.20.2/node_modules/esbuild/lib/main.js:1486:45)
    at buildResponseToResult (./node_modules/.pnpm/esbuild@0.20.2/node_modules/esbuild/lib/main.js:1057:7)
    at ./node_modules/.pnpm/esbuild@0.20.2/node_modules/esbuild/lib/main.js:1086:16
    at responseCallbacks.<computed> (./node_modules/.pnpm/esbuild@0.20.2/node_modules/esbuild/lib/main.js:704:9)
    at handleIncomingPacket (./node_modules/.pnpm/esbuild@0.20.2/node_modules/esbuild/lib/main.js:764:9)
    at Socket.readFromStdout (./node_modules/.pnpm/esbuild@0.20.2/node_modules/esbuild/lib/main.js:680:7)
    at Socket.emit (node:events:517:28)
    at addChunk (node:internal/streams/readable:368:12)

WARN Broken build, fix the error above.
WARN You may need to refresh the browser.

 ELIFECYCLE  Command failed with exit code 1.
 ```
 
The moment I add `@storybook/core` as a dependency to the project, it starts working!

Reply:
The issue above is that `esbuild` tries to resolve from the root. And the root does not have `@storybook/core`.

I'm considering making some changes so it does actually work, though possibly that would cause the user not to notice they have to add the dependency to `@storybook/core`?

I'm going to have to improve the CLI check somehow.